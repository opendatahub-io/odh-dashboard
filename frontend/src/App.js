import React from 'react';
import '@patternfly/react-core/dist/styles/base.css';
import {
  Page,
} from '@patternfly/react-core';
import Launcher from './Launcher/Launcher.js'
import Header from './Header.js'

  /**
   * This function wraps up the output from Launcher.js into a React Fragment and returns to the index page
   */
function App() {
  return (
    <React.Fragment>
    <Page
          header={Header}
        >
    <Launcher />      
    </Page>
    </React.Fragment>
      )}

export default App;
