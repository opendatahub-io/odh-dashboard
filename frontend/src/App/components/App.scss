.app {
  .header {
    height: 36px;
  }
}
