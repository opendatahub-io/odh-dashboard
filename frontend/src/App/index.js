export * from "./components/App";
export * from "./reducer";
export { App as default } from "./components/App.js";
