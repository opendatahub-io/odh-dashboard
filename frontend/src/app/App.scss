html, body, #root {
  height: 100%;
}

#dashboard-page-main {
  // Establish positioning relative to the top left of the main container.
  position: relative;
}

.odh-dashboard {
  &__page-content {
    border: 1px solid var(--pf-global--BorderColor--100);
    display: flex;
    flex: 1;
    margin: var(--pf-global--spacer--md);
  }
  .pf-c-app-launcher__group-title {
    font-size: 13px;
  }
  .pf-c-app-launcher__menu-item-external-icon {
    opacity: 1;
    color: var(--pf-global--icon--Color--light);
  }
  &__brand {
    height: 36px;
  }
}
// specificity targeting form elements to override --pf-global--FontSize--md
.pf-c-page, .pf-c-modal-box {
  .pf-c-app-launcher,
  .pf-c-button,
  .pf-c-dropdown,
  .pf-c-dropdown__menu-item,
  .pf-c-dropdown__toggle,
  .pf-c-select,
  .pf-c-select__menu-item,
  .pf-c-select__toggle,
  .pf-c-text-input-group,
  .pf-c-form-control {
    font-size: var(--pf-global--FontSize--sm);
    height: auto;
  }
}
