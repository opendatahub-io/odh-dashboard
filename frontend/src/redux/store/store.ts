import { createStore, applyMiddleware, compose } from 'redux';
import reduxThunk from 'redux-thunk';
import rootReducer from '../reducers/rootReducers';

// eslint-disable-next-line
const composeEnhancers = (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const configureStore = () => {
  return createStore(rootReducer, composeEnhancers(applyMiddleware(reduxThunk)));
};

export const store = configureStore();
