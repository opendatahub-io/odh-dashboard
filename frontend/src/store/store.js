import { createStore, applyMiddleware, compose } from "redux";
import reduxThunk from "redux-thunk";
import rootReducer from "./reducers";

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const configureStore = () => {
  return createStore(rootReducer, composeEnhancers(applyMiddleware(reduxThunk)));
};

export const store = configureStore();
