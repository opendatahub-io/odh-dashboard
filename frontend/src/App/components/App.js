import React, { useState, useEffect } from "react";
import { withRouter } from "react-router-dom";
import { connect } from "react-redux";

import { Page } from "@patternfly/react-core";
// import { Header, NavSidebar, Routes } from "./";
import { Header, Routes } from "./";
import { getComponents } from "../actions";

import "./App.scss";

const _App = withRouter(({ getComponents }) => {
  const [isNavOpen, setIsNavOpen] = useState(true);

  useEffect(() => {
    getComponents();
  }, []);

  const onNavToggle = () => {
    setIsNavOpen(!isNavOpen);
  };

  return (
    <Page
      header={<Header isNavOpen={isNavOpen} onNavToggle={onNavToggle} />}
      // sidebar={<NavSidebar isNavOpen={isNavOpen} />}
      isManagedSidebar
      className="app"
    >
      <Routes />
    </Page>
  );
});

const mapStateToProps = (state) => {
  return state.appReducer;
};

const mapDispatchToProps = (dispatch) => ({
  getComponents: () => {
    dispatch(getComponents());
  },
});

export const App = connect(mapStateToProps, mapDispatchToProps)(_App);
export default App;
