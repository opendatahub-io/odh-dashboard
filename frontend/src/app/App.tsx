import React from 'react';
import '@patternfly/patternfly/patternfly.min.css';
import '@patternfly/patternfly/patternfly-addons.css';
import {
  Alert,
  Bullseye,
  Button,
  Page,
  PageSection,
  Spinner,
  Stack,
  StackItem,
} from '@patternfly/react-core';
import ErrorBoundary from '~/components/error/ErrorBoundary';
import ToastNotifications from '~/components/ToastNotifications';
import { useWatchBuildStatus } from '~/utilities/useWatchBuildStatus';
import { useUser } from '~/redux/selectors';
import { DASHBOARD_MAIN_CONTAINER_SELECTOR } from '~/utilities/const';
import useDetectUser from '~/utilities/useDetectUser';
import ProjectsContextProvider from '~/concepts/projects/ProjectsContext';
import Header from './Header';
import AppRoutes from './AppRoutes';
import NavSidebar from './NavSidebar';
import AppNotificationDrawer from './AppNotificationDrawer';
import { AppContext } from './AppContext';
import { useApplicationSettings } from './useApplicationSettings';
import TelemetrySetup from './TelemetrySetup';
import { logout } from './appUtils';

import './App.scss';

const App: React.FC = () => {
  const [notificationsOpen, setNotificationsOpen] = React.useState(false);
  const { username, userError, isAllowed } = useUser();

  const buildStatuses = useWatchBuildStatus();
  const {
    dashboardConfig,
    loaded: configLoaded,
    loadError: fetchConfigError,
  } = useApplicationSettings();

  useDetectUser();

  if (!username || !configLoaded || !dashboardConfig) {
    // We lack the critical data to startup the app
    if (userError || fetchConfigError) {
      // There was an error fetching critical data
      return (
        <Page>
          <PageSection>
            <Stack hasGutter>
              <StackItem>
                <Alert variant="danger" isInline title="General loading error">
                  <p>
                    {(userError ? userError.message : fetchConfigError?.message) ||
                      'Unknown error occurred during startup.'}
                  </p>
                  <p>Logging out and logging back in may solve the issue.</p>
                </Alert>
              </StackItem>
              <StackItem>
                <Button
                  variant="secondary"
                  onClick={() => logout().then(() => window.location.reload())}
                >
                  Logout
                </Button>
              </StackItem>
            </Stack>
          </PageSection>
        </Page>
      );
    }

    // Assume we are still waiting on the API to finish
    return (
      <Bullseye>
        <Spinner />
      </Bullseye>
    );
  }

  return (
    <AppContext.Provider
      value={{
        buildStatuses,
        dashboardConfig,
      }}
    >
      <Page
        className="odh-dashboard"
        isManagedSidebar
        header={<Header onNotificationsClick={() => setNotificationsOpen(!notificationsOpen)} />}
        sidebar={isAllowed ? <NavSidebar /> : undefined}
        notificationDrawer={<AppNotificationDrawer onClose={() => setNotificationsOpen(false)} />}
        isNotificationDrawerExpanded={notificationsOpen}
        mainContainerId={DASHBOARD_MAIN_CONTAINER_SELECTOR}
      >
        <ErrorBoundary>
          <ProjectsContextProvider>
            <AppRoutes />
          </ProjectsContextProvider>
          <ToastNotifications />
          <TelemetrySetup />
        </ErrorBoundary>
      </Page>
    </AppContext.Provider>
  );
};

export default App;
