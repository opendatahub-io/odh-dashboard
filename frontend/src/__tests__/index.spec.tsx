import * as React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { store } from '../redux/store/store';
import { BrowserRouter as Router } from 'react-router-dom';
import App from '../app/App';

jest.mock('react-router-dom', () => {
  const reactRouterDom = jest.requireActual('react-router-dom');
  return {
    ...reactRouterDom,
    useLocation: () => '/',
  };
});
jest.mock('react', () => {
  const React = jest.requireActual('react');
  jest.mock('react', () => {
    const React = jest.requireActual('react');
    return {
      ...React,
      Suspense: ({ children }) => children,
      lazy: (factory) => factory(),
    };
  });

  return React;
});

describe('Index test', () => {
  it('should render a basic component', () => {
    const component = mount(
      <Provider store={store}>
        <Router>
          <App />
        </Router>
      </Provider>,
    );
    expect(component.html()).toMatchSnapshot('basic');
  });
});
