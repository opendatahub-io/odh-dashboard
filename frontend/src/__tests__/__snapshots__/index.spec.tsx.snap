// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index test should render a basic component: basic 1`] = `"<div class=\\"pf-c-page odh-dashboard\\"><header class=\\"pf-c-page__header\\"><div class=\\"pf-c-page__header-brand\\"><div class=\\"pf-c-page__header-brand-toggle\\"><button id=\\"nav-toggle\\" aria-expanded=\\"true\\" aria-disabled=\\"false\\" aria-label=\\"Global navigation\\" class=\\"pf-c-button pf-m-plain\\" type=\\"button\\" data-ouia-component-type=\\"PF4/Button\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Button-plain-1\\"><svg style=\\"vertical-align: -0.125em;\\" fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 448 512\\" aria-hidden=\\"true\\" role=\\"img\\"><path d=\\"M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z\\"></path></svg></button></div><a class=\\"pf-c-page__header-brand-link\\"><img class=\\"pf-c-brand\\" src=\\"[object Object]\\" alt=\\"ODH Logo\\"></a></div><div class=\\"pf-c-page__header-tools\\"><div class=\\"pf-c-page__header-tools-group hidden-xs\\"><div class=\\"pf-c-page__header-tools-item\\"><button aria-disabled=\\"false\\" class=\\"pf-c-button pf-m-plain\\" type=\\"button\\" data-ouia-component-type=\\"PF4/Button\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Button-plain-2\\"><span class=\\"pf-c-notification-badge pf-m-read\\"><svg style=\\"vertical-align: -0.125em;\\" fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 896 1024\\" aria-hidden=\\"true\\" role=\\"img\\"><path d=\\"M448,0 C465.333333,0 480.333333,6.33333333 493,19 C505.666667,31.6666667 512,46.6666667 512,64 L512,106 L514.23,106.45 C587.89,121.39 648.48,157.24 696,214 C744,271.333333 768,338.666667 768,416 C768,500 780,568.666667 804,622 C818.666667,652.666667 841.333333,684 872,716 C873.773676,718.829136 875.780658,721.505113 878,724 C890,737.333333 896,752.333333 896,769 C896,785.666667 890,800.333333 878,813 C866,825.666667 850.666667,832 832,832 L63.3,832 C44.9533333,831.84 29.8533333,825.506667 18,813 C6,800.333333 0,785.666667 0,769 C0,752.333333 6,737.333333 18,724 L24,716 L25.06,714.9 C55.1933333,683.28 77.5066667,652.313333 92,622 C116,568.666667 128,500 128,416 C128,338.666667 152,271.333333 200,214 C248,156.666667 309.333333,120.666667 384,106 L384,63.31 C384.166667,46.27 390.5,31.5 403,19 C415.666667,6.33333333 430.666667,0 448,0 Z M576,896 L576,897.08 C575.74,932.6 563.073333,962.573333 538,987 C512.666667,1011.66667 482.666667,1024 448,1024 C413.333333,1024 383.333333,1011.66667 358,987 C332.666667,962.333333 320,932 320,896 L576,896 Z\\"></path></svg></span></button></div><div class=\\"pf-c-page__header-tools-item\\"><div class=\\"pf-c-dropdown pf-m-align-right\\" data-ouia-component-type=\\"PF4/Dropdown\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Dropdown-1\\"><button data-ouia-component-type=\\"PF4/DropdownToggle\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-DropdownToggle-1\\" id=\\"toggle-id\\" class=\\"pf-c-dropdown__toggle\\" type=\\"button\\" aria-expanded=\\"false\\" aria-haspopup=\\"true\\"><span class=\\"pf-c-dropdown__toggle-text\\"><svg style=\\"vertical-align: -0.125em;\\" fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 512 512\\" aria-hidden=\\"true\\" role=\\"img\\"><path d=\\"M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zM262.655 90c-54.497 0-89.255 22.957-116.549 63.758-3.536 5.286-2.353 12.415 2.715 16.258l34.699 26.31c5.205 3.947 12.621 3.008 16.665-2.122 17.864-22.658 30.113-35.797 57.303-35.797 20.429 0 45.698 13.148 45.698 32.958 0 14.976-12.363 22.667-32.534 33.976C247.128 238.528 216 254.941 216 296v4c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12v-1.333c0-28.462 83.186-29.647 83.186-106.667 0-58.002-60.165-102-116.531-102zM256 338c-25.365 0-46 20.635-46 46 0 25.364 20.635 46 46 46s46-20.636 46-46c0-25.365-20.635-46-46-46z\\"></path></svg></span><span class=\\"pf-c-dropdown__toggle-icon\\"><svg style=\\"vertical-align: -0.125em;\\" fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 320 512\\" aria-hidden=\\"true\\" role=\\"img\\"><path d=\\"M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z\\"></path></svg></span></button></div></div><div class=\\"pf-c-page__header-tools-item\\"><div class=\\"pf-c-dropdown pf-m-align-right\\" data-ouia-component-type=\\"PF4/Dropdown\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Dropdown-2\\"><button data-ouia-component-type=\\"PF4/DropdownToggle\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-DropdownToggle-2\\" id=\\"toggle-id\\" class=\\"pf-c-dropdown__toggle\\" type=\\"button\\" aria-expanded=\\"false\\" aria-haspopup=\\"true\\"><span class=\\"pf-c-dropdown__toggle-icon\\"><svg style=\\"vertical-align: -0.125em;\\" fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 320 512\\" aria-hidden=\\"true\\" role=\\"img\\"><path d=\\"M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z\\"></path></svg></span></button></div></div></div></div></header><div id=\\"page-sidebar\\" class=\\"pf-c-page__sidebar pf-m-expanded\\" aria-hidden=\\"false\\"><div class=\\"pf-c-page__sidebar-body\\"><nav class=\\"pf-c-nav nav\\" aria-label=\\"Nav\\" data-ouia-component-type=\\"PF4/Nav\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-Nav-1\\"><ul class=\\"pf-c-nav__list\\"><li class=\\"pf-c-nav__item pf-m-expandable\\" data-ouia-component-type=\\"PF4/NavExpandable\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-NavExpandable-1\\" aria-label=\\"Applications\\"><button class=\\"pf-c-nav__link\\" id=\\"apps\\" aria-expanded=\\"false\\">Applications<span class=\\"pf-c-nav__toggle\\"><span class=\\"pf-c-nav__toggle-icon\\"><svg style=\\"vertical-align: -0.125em;\\" fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" viewBox=\\"0 0 256 512\\" aria-hidden=\\"true\\" role=\\"img\\"><path d=\\"M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z\\"></path></svg></span></span></button><section class=\\"pf-c-nav__subnav\\" aria-labelledby=\\"apps\\" hidden=\\"\\"><ul class=\\"pf-c-nav__list\\"><li class=\\"pf-c-nav__item\\" data-ouia-component-type=\\"PF4/NavItem\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-NavItem-1\\"><a aria-label=\\"Enabled\\" class=\\"pf-c-nav__link\\" href=\\"/\\">Enabled</a></li><li class=\\"pf-c-nav__item\\" data-ouia-component-type=\\"PF4/NavItem\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-NavItem-2\\"><a aria-label=\\"Explore\\" class=\\"pf-c-nav__link\\" href=\\"/explore\\">Explore</a></li></ul></section></li><li class=\\"pf-c-nav__item\\" data-ouia-component-type=\\"PF4/NavItem\\" data-ouia-safe=\\"true\\" data-ouia-component-id=\\"OUIA-Generated-NavItem-3\\"><a aria-label=\\"Resources\\" class=\\"pf-c-nav__link\\" href=\\"/resources\\">Resources</a></li></ul></nav></div></div><div class=\\"pf-c-page__drawer\\"><div class=\\"pf-c-drawer\\"><div class=\\"pf-c-drawer__main\\"><div class=\\"pf-c-drawer__content\\"><div class=\\"pf-c-drawer__body\\"><main class=\\"pf-c-page__main\\" tabindex=\\"-1\\"><section class=\\"pf-c-page__main-section pf-m-light odh-apps__heading\\"><div class=\\"pf-c-content odh-apps__heading__text\\"><h1 data-pf-content=\\"true\\" class=\\"\\"></h1><p data-pf-content=\\"true\\" class=\\"\\"></p></div></section><section class=\\"pf-c-page__main-section pf-m-fill\\"><div class=\\"pf-c-empty-state\\" data-test-id=\\"loading-empty-state\\"><div class=\\"pf-c-empty-state__content\\"><span class=\\"pf-c-spinner pf-m-xl\\" role=\\"progressbar\\" aria-valuetext=\\"Loading...\\"><span class=\\"pf-c-spinner__clipper\\"></span><span class=\\"pf-c-spinner__lead-ball\\"></span><span class=\\"pf-c-spinner__tail-ball\\"></span></span><h5 class=\\"pf-c-title pf-m-lg\\">Loading</h5></div></div></section><div class=\\"odh-dashboard__notifications\\"></div></main></div></div><div class=\\"pf-c-drawer__panel\\" hidden=\\"\\"></div></div></div></div></div>"`;
